odoo/sql_db.py

```diff
@@ -61,18 +63,11 @@ def adapt_string(adapted):
 SENTINEL = object()
 
 
-def get_env(frame, back=0):
-    """ Retrieve an environment from the given stack frame. """
-    for i in range(back):
-        frame = frame.f_back
-    while frame:
-        env = frame.f_locals.get('env', SENTINEL)
-        if env is not SENTINEL:
-            return env
-        env = getattr(frame.f_locals.get('self'), 'env', SENTINEL)
-        if env is not SENTINEL:
+def get_env(cr):
```

Comment:
Now that `get_env()` uses `cr` you might want to make it a method to avoid the extra param. Or keep it as a helper... either is fine for me :)